internal/services/devtestlabs/dev_test_virtual_network_resource.go

```diff
@@ -330,9 +396,45 @@ func flattenDevTestVirtualNetworkSubnets(input *[]virtualnetworks.SubnetOverride
 		}
 		output["use_public_ip_address"] = v.UsePublicIPAddressPermission
 		output["use_in_virtual_machine_creation"] = v.UseInVMCreationPermission
+		output["shared_public_ip_address"] = flattenDevTestVirtualNetworkSubnetIpAddressConfiguration(v.SharedPublicIPAddressConfiguration)
 
 		outputs = append(outputs, output)
 	}
 
 	return outputs
 }
+
+func flattenDevTestVirtualNetworkSubnetIpAddressConfiguration(input *virtualnetworks.SubnetSharedPublicIPAddressConfiguration) []interface{} {
+	outputs := make([]interface{}, 0)
+
+	if input == nil {
+		return outputs
+	}
+
+	output := make(map[string]interface{})
+	if input.AllowedPorts != nil {
+		output["allowed_ports"] = flattenDevTestVirtualNetworkSubnetAllowedPorts(input.AllowedPorts)
+	}
+	outputs = append(outputs, output)
+	return outputs
+}
+
+func flattenDevTestVirtualNetworkSubnetAllowedPorts(input *[]virtualnetworks.Port) []interface{} {
+	outputs := make([]interface{}, 0)
+	if input == nil {
+		return outputs
+	}
+
+	for _, v := range *input {
+		output := make(map[string]interface{})
+		if v.BackendPort != nil {
+			output["backend_port"] = v.BackendPort
+		}
+		if v.TransportProtocol != nil {
+			output["transport_protocol"] = v.TransportProtocol
+		}
```

Comment:
These could be condensed using `pointer.From`
```suggestion
		output["backend_port"] = pointer.From(v.BackendPort)
		output["transport_protocol"] = pointer.From(v.TransportProtocol)
```

Reply:
Will change.